test/src/unit-items.cpp

```diff
@@ -39,6 +39,18 @@ using nlohmann::json;
     #define JSON_HAS_CPP_14
 #endif
 
+#ifdef DOCTEST_PLATFORM_WINDOWS
+#if DOCTEST_CLANG && DOCTEST_CLANG >= DOCTEST_COMPILER(10, 0, 0)
+    DOCTEST_CLANG_SUPPRESS_WARNING("-Wrange-loop-construct")
+#endif
+#endif
+
+#ifdef DOCTEST_PLATFORM_MAC
+#if DOCTEST_CLANG && DOCTEST_CLANG >= DOCTEST_COMPILER(12, 0, 0)
+    DOCTEST_CLANG_SUPPRESS_WARNING("-Wrange-loop-analysis")
+#endif
+#endif
+
```

Comment:
@nlohmann Any idea why the MacOSX part here does not work? https://travis-ci.org/github/nlohmann/json/jobs/707957810 still fails.

Reply:
This is due to https://github.com/onqtam/doctest/issues/415, I have a workaround.